### DISCUSSION TITLE
MQTT rc-2

### DISCUSSION THREAD

**Discussion by Tom618s** [arendst/Tasmota#15382] [Q&A] [answered]
This issue is from version 9.3.0
 If you download version 9.1.0 this problem still did not exist
 For those who are having this problem just download the older version
![Screenshot_2022-04-15-22-10-59-246_com android chrome](https://user-images.githubusercontent.com/95147431/163673250-4e863db7-3f0b-4aac-a075-f7f03de1d27a.jpg)

**Comment by barbudor** [comment]
If you have a problem, it is because you are doing something wrong.
![image](https://user-images.githubusercontent.com/2996042/163673839-2678e5a8-c5af-40ed-aa6c-ab46e8517d39.png)

Error -2 means that your device cannot even connect to the server. This is a network problem.

It took me 5 minutes to discover how to configure Tasmota 11.1.0 on Smarnest.cz
```
13:56:16.059 MQT: Attempting connection...
13:56:16.147 MQT: Connected
13:56:16.150 MQT: 625****************edd/tele/LWT = Online (retained)
13:56:16.152 MQT: 625****************edd/cmnd/POWER = 
13:56:16.154 MQT: Subscribe to 625****************edd/cmnd/#
13:56:16.157 MQT: Subscribe to tasmotas/cmnd/#
13:56:16.161 MQT: Subscribe to cmnd/625****************edd_fb/#
13:56:16.169 MQT: 625****************edd/tele/INFO1 = {"Info1":{"Module":"Generic","Version":"11.1.0.1(bhome)","FallbackTopic":"cmnd/625****************edd_fb/","GroupTopic":"tasmotas/cmnd/"}}
13:56:16.187 MQT: 625****************edd/tele/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"nodemcu","IPAddress":"192.168.168.195"}}
13:56:16.199 MQT: 625****************edd/tele/INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":7}}
```

**Reply by sfromis** [reply]
Looks like smartnest.cz is recommending to use unencrypted MQTT over the Internet, instead of MQTT-TLS. From a security viewpoint, that's not great. Anyway, maybe your router or ISP has restrictions about what to allow.

**Comment by barbudor** [comment, marked as answer]
It appears that SMARTNEST requires Client and Topic to use the same value
For historical reasons this was forbidden in Tasmota since a very long time (even before v6).
And for some reasons, there was a hole which is some case was letting this to happen (probably related to the order in which the commands were passed in the console).

This constraint has been droppped in v10.1.0.1 so any recent version doesn't have the problem anymore

Please try a recent version like the just released v11.1.0

**Reply by Tom618s** [reply]
Eu não acredito que tenha feito algo de errado, sem mudar nada, apenas fiz a atualização via OTA para a versão mais antiga e ele simplesmente conectou devido as configurações MQTT que estavam salvas no dispositivo

![Screenshot_2022-04-10-14-34-15-663_com android chrome](https://user-images.githubusercontent.com/95147431/163678119-455cb9a6-0e0a-4e3f-973f-b389b36bec47.jpg)
. Eu não entendo sobre os tópicos serem os mesmos, mas aqui está as configurações salvas

**Reply by barbudor** [reply]
@Tom618s 
Please use English on this forum
You are still not using the right version:  10.1.0 is _before_ 10.1.0.1
Just jump to latest 11.1.0

**Reply by Tom618s** [reply]
![Screenshot_2022-04-16-14-04-07-673_com android chrome](https://user-images.githubusercontent.com/95147431/163684551-7a6c4631-4c5d-41e9-bcd7-1a06bc05b8ec.jpg)
Here it is, the same configuration
MQTT in version 11.1

**Reply by Tom618s** [reply]
I confess that I don't understand much about the subject, but I just know that changing to version 9.1.0 the problem does not appear

**Reply by Tom618s** [reply]
![Screenshot_2022-04-16-14-14-19-153_com android chrome](https://user-images.githubusercontent.com/95147431/163684831-52477c9d-10e5-4887-9027-94d9adb33336.jpg)

the connection was successful just changing the version

**Reply by barbudor** [reply]
Instead of posting those big pictures, please just copy/paste the logs as text from the web console, using the code tag (the `< >` icon in the tool bar of Github)
![image](https://user-images.githubusercontent.com/2996042/163686256-4c0f6a56-f7c3-4b82-a6e7-ad9a4974d287.png)

As I proved above, I've been able to connect a NodeMCU to Smartnest.cz using latest version
Beside error: -2 is a network a network error : Tasmota cannot reach the server

What is your device ? ESP8266 or ESP32 ?
When using v11.1, what is the result of command `WebSend [smartnest.cz] /`
You should get `MQT: stat/nodemcu/WEBSEND = {"WebSend":"Done"}` otherwise that means that somehting is wrong in your network configuration.
May be checking `IpAddress` where IpAddress2 should be the IP address of your Internet router and IpAddress4 should be a valid DNS server

**Reply by barbudor** [reply]
Looking back at the changes between 9.1.0 and 9.3.0, there has been a change in the MQTT Wifi connection timeout. Originally it was 5 sec but it was reduced to 200ms
200 ms may not be enough in your case depending where you are located vs smartnest.cz 
Despite the CZ extension, it looks like their servers are located in Frankfurt, Germany

Can you please try `MqttWifiTimeout 1000` ? or even higher if needed

**Reply by Tom618s** [reply]
Thank you very much, thank you very much, that was exactly the problem, after I put MqttWifiTimeout 1000 the connection worked. I am very grateful for your help

**Reply by barbudor** [reply]
You're welcomed 😊

**Comment by jfery63** [comment]
Hello,
I had exactly the same problem, to solve the problem I went to the file /etc/mosquitto/mosquitto.conf and I added this line at the end of the file
listener 1883
And miraculously, everything works.
John

**Reply by barbudor** [reply]
@jfery63 Not the same problemm at all as the author is using a cloud based MQTT broker which is obviously listenning to 1883.

**Reply by jfery63** [reply]
ah ok i thought it was on a raspbery, well if it can help someone... 
because i searched for hours to find out why it didn't work and i found this solution.
John
